test/unit_test.c

```diff
@@ -2127,8 +2144,18 @@ static void test_str(void) {
     TESTDOUBLE("%g", 10000.0, "10000");
     TESTDOUBLE("%g", 100000.0, "100000");
     TESTDOUBLE("%g", 1000000.0, "1e+06");
+    TESTDOUBLE("%f", 1000000.0, "1000000.000000");
     TESTDOUBLE("%g", 10000000.0, "1e+07");
+    TESTDOUBLE("%f", 10000000.0, "10000000.000000");
     TESTDOUBLE("%g", 100000001.0, "1e+08");
+    TESTDOUBLE("%g", 0.1, "0.1");
+    TESTDOUBLE("%g", 0.01, "0.01");
+    TESTDOUBLE("%g", 0.001, "0.001");
+    TESTDOUBLE("%g", 0.0001, "0.0001");
+    TESTDOUBLE_NOHOSTCHECK("%g", 0.00001, "0.00001"); // "1e-05"
+    TESTDOUBLE("%g", 0.000001, "1e-06");
+    TESTDOUBLE("%g", -0.0001, "-0.0001");
+    TESTDOUBLE_NOHOSTCHECK("%g", -0.00001, "-0.00001"); // "-1e-05"
```

Comment:
The NOHOSTCHECK tests are added cause the result is implementation specific, and not stable across platforms

Reply:
I explicitly added those two I'm marking here, please see I added an extra commit to this PR. I did this because I knew we had some issues before and wanted to check.
Those tests I added fail, unless I mark them as NOHOSTCHECK because

_AFAICS we are expected to print **1e-05** and yet we print **0.00001**_

So either we:
- make them work as Linux/? do
- let them as they are until someone complains (if ever)

Reply:
Let's go for the latter ? 
Since this PR makes the code less broken anyway.

Reply:
@scaprile could you take over this please?